Author: Lionel Elie Mamane <lionel@mamane.lu>
Description: React sensibly on integer overflow in date handling (Closes: #326024, CVE-2005-4153)
Bug-Debian: #326024
Forwarded: no

Index: b/Mailman/Handlers/Scrubber.py
===================================================================
--- a/Mailman/Handlers/Scrubber.py	2010-05-24 17:51:12.000000000 +0200
+++ b/Mailman/Handlers/Scrubber.py	2010-05-24 17:51:13.000000000 +0200
@@ -122,7 +122,7 @@
                      }.get(parts[3], 0)
             day = int(parts[4])
             year = int(parts[6])
-        except (IndexError, ValueError):
+        except (OverflowError, IndexError, ValueError):
             # Best we can do I think
             month = day = year = 0
         datedir = '%04d%02d%02d' % (year, month, day)
